### PR TITLE
Move app_wrapper deface to spree/layouts/admin view

### LINKED ISSUE #2807: [Defacepocalypse] Convert heavily defaced Spree views to OFN views
This EPIC represents the technical initiative to move away from deface and implement [this convention](https://github.com/openfoodfoundation/openfoodnetwork/wiki/Code,-the-way-we-do-things#re-building-spree-views).
Some of this work will be done as part of the spree upgrade, but not necessarily.

### LINKED ISSUE #4050: Make OFN independent of spree_backend
### What we should change and why (this is tech debt)
Most of spree_backend code is overriden in OFN thus creating a mess between OFN code on top of spree code. By removing spree_backend and moving what's needed to OFN, working with our code base becomes a LOT easier.

Before we do this we need to finish de-deface all spree_backend views as part of #2807, this is very close to completion now :tada:

After this is done, we can move the views and controllers that are not in OFN yet:
- 8 in the config space:
  - - [X] countries, states and zones #4290
  - - [X] general settings, image_settings and mail methods #4289
  - - [X] tax_categories, tax_rates and tax_settings #4292
  - - [X] taxonomies and taxons #4291
  - - [X] shipping_methods and shipping_categories #4269
  - - [X] payment_methods (merge controllers) #4371
- 5 in the product space:
  - - [X] images (copy controller and views to OFN) #4372
  - - [x] option_types and option_values (copy controllers and views to OFN) moved in #4389 but removed in #4397
  - - [x]  prototypes - the prototypes feature is not used since 2013 #4395
  - - [x] product_properties and properties - (copy properties views and the controllers) #4449
- 1 in the orders space:
  - - [x] return_authorizations (copy views and controller) #4450
- and 1 for user searches:
  - - [x] search (merge controller and get rid of rabl view, is this really used in the users admin page?) #4511

These are the 9 views we dont need to copy because we dont use them in OFN: banners, promotion_actions, promotion_rules, promotions, stock_items, stock_locations, stock_movements, stock_transfers, trackers (we should remove GA support).

We have 10 controller decorators that need to be merged with the original controllers (all code in original controller that is not needed should not be copied):
  - - [x] adjustments_controller_decorator.rb #4515
  - - [x] base_controller_decorator.rb #4512
  - - [x] orders_controller_decorator.rb #4513
  - - [x] customer_details_controller_decorator.rb #4513
  - - [x] overview_controller_decorator.rb #4514
  - - [x] payments_controller_decorator.rb #4509
  - - [x] products_controller_decorator.rb #4508
  - - [x] variants_controller_decorator.rb #4508
  - - [x] reports_controller_decorator.rb #4507
  - - [x] resource_controller_decorator.rb #4510
  - - [x] search_controller_decorator.rb #4511

And we have 3 controllers that are not decorated in OFN but may be needed, or maybe not:
  - - [x] line_items_controller.rb (not needed)
  - - [x] stock_items_controller.rb (not needed)
  - - [x] inventory_units_controller.rb (not needed)

Spree views in spree/admin/shared that may be needed:
  - - [x] _destroy.js.erb #4588
  - - [x] _new_resource_links.html.erb  #4588
  - - [x] _translations.html.erb  #4588
  - - [x] _address_form.html.erb #4513
  - - [x] _update_order_state.js #4513
  - - [x] _calculator_fields.html.erb #4585
  - - [x] _address.html.erb (not needed)
  - - [x] _alert.html.erb (not needed)
  - - [x] _order_details.html.erb (not needed)
  - - [x] _report_criteria.html.erb (not needed)
  - - [x] _show_resource_links.html.erb (not needed)

The trickiest part of this task (but also one of the best!) will probably be the clean up of the JS code under /assets as we should only copy what is needed. I reckon not much will be copied. This will be tricky because this is not angular code.

Other things that need to be copied: CSS, helpers and routes.

### Impact and timeline
There are a few hundreds of lines of code that need to be moved around but just imagine how much better the OFN codebase will look after this, not a single controller_decorator!!!!

### PR DESCRIPTION
**Pull request by luisramos0**
Not an easy PR but I think this is nailing it.

#### What? Why?

Together with #3615, #4074 and #4164, this PR will actually close #2807 🎉 
All other PRs mentioned are merged now so:
Closes #2807 🎉 

#### What should we test?
We need to run very basic tests on the following pages in the backoffice (if the ng-app is not present the pages will fail completely): Customers, Enterprises, OrderCycles, Subscriptions, Orders management, Bulk Order Management and order customer details edit page.

#### Release notes
Changelog Category: Changed
Removed the very last bit of defaced html views from OFN.

### CODE REVIEW

**Comment by sigmundpetersen** [comment]
✅

File: app/views/spree/layouts/admin.html.haml [hunk 1]
```diff
@@ -4,80 +4,12 @@
...
-      = raw "Spree.api_key = \"#{try_spree_current_user.try(:spree_api_key).to_s}\";"
+    - if content_for?(:main_ng_app_name)
+      - if content_for?(:main_ng_ctrl_name)
+        %div{ "ng-app" => yield(:main_ng_app_name).strip.html_safe, "ng-controller" => yield(:main_ng_ctrl_name).strip.html_safe }
+          = render :partial => 'spree/layouts/admin_body'
+      - else
+        %div{ "ng-app" => yield(:main_ng_app_name).strip.html_safe }
+          = render :partial => 'spree/layouts/admin_body'
+    - else
+      = render :partial => 'spree/layouts/admin_body'
```
Comment (luisramos0) [inline app/views/spree/layouts/admin.html.haml]: I tried a few times and failed to make this look better...

Comment (sauloperez) [reply app/views/spree/layouts/admin.html.haml]: I think is the least of the problems. It's a negligible downside for an incredible upside this PR brings

**Comment by sauloperez** [comment]
`app/overrides/` gone! I can't believe it!

**Comment by daniellemoorhead** [comment]
>We need to run very basic tests on the following pages in the backoffice (if the ng-app is not present the pages will fail completely): Customers, Enterprises, OrderCycles, Subscriptions, Orders management, Bulk Order Management and order customer details edit page.

@luisramos0 I assume that basic tests aren't just going to the pages to make sure that we can access them, but that we can complete changes/tasks on these pages?

**Comment by daniellemoorhead** [comment]
@RachL @filipefurtad0 I'm deploying this to Katuma staging, but I don't think I'll get time to test it today. So it will be ready for someone else to pick up while I'm sleeping :)

**Comment by luisramos0** [comment]
yeah, simple tasks.

**Comment by Matt-Yorkley** [comment]
Does this mean we no longer use any Spree views directly, or just that we no longer _override_ any Spree views?

**Comment by luisramos0** [comment]
for now just the second one Matt :-) we no longer override any Spree views

#4050 is the "we no longer use any Spree views" part

**Comment by filipefurtad0** [comment]
The basic functionality works fine. But I am getting a strange behavior on
Reports > Order Cycle Management
https://staging.katuma.org/admin/reports/order_cycle_management

First, the red letters "Sorry something went wrong", and now incomplete/strange formatting. This is reproducible, after logging out and in again. Other reports seem to work well.

https://drive.google.com/open?id=1q8HdE0s_PM5mO8a1KFnXF5SxXU24S4oaIa4U8xWgdCY

**Comment by luisramos0** [comment]
@filipefurtad0 this will not affect reports either.
I think this issue is ready to go.

In the testing notes I dont see the error on the reports page... what do you mean by weird format? maybe that is also a case for a separate issue.

**Comment by filipefurtad0** [comment]
The message in red only appeared once.. and I got no screenshot, shame on me :-/ and I couldn't get it show up again.

The weird table format I mean seemed reproducible (below on the pics). It doenst look right. Should I re-stage on Katuma, just to make sure this is reproducible?

**Comment by RachL** [comment]
@filipefurtad0 this is particular to katuma staging. Sorry I forgot to tell you about it. Katuma indeed did an override of this report and most of the time the override is not properly activated in staging. If you are not sure you can restage the pr on another staging server and see that this report will work :)

**Comment by sauloperez** [comment]
My apologies @filipefurtad0 . That's a :hankey: we want to remove but I haven't had the time to deal with it yet.

**Comment by filipefurtad0** [comment]
Hey, no problem at all @RachL @sauloperez :-) When something like this comes up I'll restage it somewhere else, to make sure. :+1:

**Pull request merged**